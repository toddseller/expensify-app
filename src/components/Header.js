import { NavLink } from 'react-router-dom'
import React from 'react'

export default class Header extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      isAuthenticated: true
    }
  }

  render() {
    return (
      <div>
        { this.state.isAuthenticated &&
          <header>
            <h1>Expensify</h1>
            <NavLink to="/" activeClassName="is-active" exact={ true }>Dashboard&nbsp;&nbsp;</NavLink>
            <NavLink to="/create" activeClassName="is-active">Create Expense&nbsp;&nbsp;</NavLink>
          </header>
        }
      </div>
    )
  }
}