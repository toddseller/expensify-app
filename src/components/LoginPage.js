import React from 'react'
import { connect } from 'react-redux'

import { startLogin } from '../actions/auth'

export const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expensify App</h1>
      <p>It's time to get your expenses under control.</p>
      <div className="login-button">

        <button className="login-button__button" onClick={ startLogin }><img src="/images/btn_google_light_normal_ios.svg" className="login-button__image"/><span className="login-button__text">Login with Google</span></button>
      </div>
    </div>
  </div>
)

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)