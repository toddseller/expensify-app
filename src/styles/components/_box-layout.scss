.box-layout
{
  display:         flex;
  align-items:     center;
  justify-content: center;

  width:           100vw;
  height:          100vh;

  background:      url('/images/bg.jpg');
  background-size: cover;
}

.box-layout__box
{
  width:         25rem;
  padding:       $l-size $m-size;

  text-align:    center;

  border-radius: 3px;
  background:    fade_out(white, .25);
}

.box-layout__title
{
  line-height: 1;

  margin:      0 0 $m-size 0;
}
