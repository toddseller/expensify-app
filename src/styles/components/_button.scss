.button
{
  font-size:   $font-size-large;
  font-weight: 300;

  padding:     $s-size;

  color:       white;
  border:      none;
  background:  $blue;
  display: inline-block;
  text-decoration: none;
  line-height: 1;
}

.button--link
{
  background: none;
}

.login-button
{
  display:         flex;
  justify-content: center;
}

.login-button__button
{
  display:         flex;
  align-items:     center;
  justify-content: space-evenly;

  width:           22rem;

  transition:      border .3s ease-in-out;
  text-align:      right;

  border-radius:   5px;
  background:      $google-blue;
}

.login-button__button:hover
{
  border: 1px solid darken($google-blue, 10%);
}

.login-button__image
{
  height: 5rem;
  margin: -3px 0 -4px -14px;
}

.login-button__text
{
  font-size:     $font-size-large;
  font-weight:   300;

  align-content: center;

  color:         white;
}
