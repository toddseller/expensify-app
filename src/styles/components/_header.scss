.header
{
  background: $dark-blue;
}

.header__content
{
  display:         flex;
  align-items:     center;
  justify-content: space-between;

  padding:         $s-size 0;
}

.header__title
{
  text-decoration: none;

  color:           white;
  h1
  {
    margin: 0;
  }
}
