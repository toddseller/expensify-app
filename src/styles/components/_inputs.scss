.text-input {
  border: $grey-border;
  height: 50px;
  padding: $s-size;
  font-size: $font-size-large;
  font-weight: 300;
}

.select {
  @extend .text-input;
  -webkit-appearance: none;
  -webkit-border-radius: 0;
}

.textarea {
  @extend .text-input;
  height: 10rem;
}