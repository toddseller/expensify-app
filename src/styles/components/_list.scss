.list-body
{
  margin-bottom: $m-size;

  @media (min-width: $desktop-breakpoint)
  {
    margin-bottom: $l-size;
  }
}

.list-header
{
  display: flex;
  justify-content: space-between;
  background: $off-white;
  color: $grey;
  border: $grey-border;
  padding: $s-size $m-size;
}

.list-item
{
  display: flex;
  flex-direction: column;
  border: $grey-border;
  border-top: none;
  color: $dark-grey;
  padding: $s-size;
  text-decoration: none;
  transition: background .3s ease;

  &:hover {
    background: $off-white;
  }

  @media (min-width: $desktop-breakpoint)
  {
    flex-direction: row;
    justify-content: space-between;
    padding: $m-size;
    align-items: center;
  }
}

.list-item--message
{
  align-items: center;
  justify-content: center;
  color: $grey;
  padding: $m-size;

  &:hover
  {
    background: none;
  }
}

.list-item__title
{
  margin: 0;
  word-break: all;
}

.list-item__sub-title
{
  color: $grey;
  font-size: $font-size-small;
}

.list-item__data
{
  margin: $s-size 0 0 0;

  @media (min-width: $desktop-breakpoint)
  {
    margin: 0;
    padding-left: $s-size;
  }
}