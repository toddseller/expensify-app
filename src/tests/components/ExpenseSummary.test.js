import React from 'react'
import { shallow } from 'enzyme'

import { ExpenseSummary } from '../../components/ExpenseSummary'



test('should correctly render ExpenseSummary with 1 expense', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={325}/>)
  expect(wrapper).toMatchSnapshot()
})

test('should correctly render ExpenseSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={3} expensesTotal={195035}/>)
  expect(wrapper).toMatchSnapshot()
})