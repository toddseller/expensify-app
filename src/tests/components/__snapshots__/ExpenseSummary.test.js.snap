// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $3.25
  </h1>
</div>
`;

exports[`should correctly render ExpenseSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totaling 
    $1,950.35
  </h1>
</div>
`;
